src/operations/update.ts

```diff
@@ -41,107 +46,260 @@ export interface UpdateResult {
 }
 
 /** @internal */
-export class UpdateOperation extends AbstractOperation<Document> {
-  options: UpdateOptions;
+export class UpdateOperation extends CommandOperation<Document> {
+  options: UpdateOptions & { ordered?: boolean };
   operations: Document[];
 
-  constructor(ns: MongoDBNamespace, ops: Document[], options: UpdateOptions) {
-    super(options);
+  constructor(
+    ns: MongoDBNamespace,
+    ops: Document[],
+    options: UpdateOptions & { ordered?: boolean }
+  ) {
+    super(undefined, options);
     this.options = options;
     this.ns = ns;
+
     this.operations = ops;
   }
 
   get canRetryWrite(): boolean {
+    if (super.canRetryWrite === false) {
+      return false;
+    }
+
     return this.operations.every(op => op.multi == null || op.multi === false);
   }
 
   execute(server: Server, session: ClientSession, callback: Callback<Document>): void {
-    server.update(
-      this.ns.toString(),
-      this.operations,
-      { ...this.options, readPreference: this.readPreference, session } as WriteCommandOptions,
-      callback
-    );
+    const options = this.options ?? {};
+    const ordered = typeof options.ordered === 'boolean' ? options.ordered : true;
+    const command: Document = {
+      update: this.ns.collection,
+      updates: this.operations,
+      ordered
+    };
+
+    if (typeof options.bypassDocumentValidation === 'boolean') {
+      command.bypassDocumentValidation = options.bypassDocumentValidation;
+    }
+
+    // If a command is to be explained, we need to reformat the command after
+    // the other command properties are specified.
+    // const explain = Explain.fromOptions(options);
+    // if (explain) {
+    //   command = decorateWithExplain(writeCommand, explain);
+    // }
```

Comment:
Updates can be explained so was there an issue here?

Reply:
the command is actually decorated with explain in the `CommandOperation#executeCommand` method